### DISCUSSION TITLE
current working dir

### DISCUSSION THREAD

**Discussion by gswifort** [CEXT-Dan/PyRx#38] [Ideas]
When importing modules (pyload), `cwd` is changed to the directory of this module. It would be useful to have a constant, defined during loading `.arx`, which would store the original `cwd`, i.e. the directory from which Autocad was launched.

**Comment by CEXT-Dan** [comment]
Yes, AutoCAD is always changing it, I.e. during open file dialog and during commands.
BricsCAD and the other clones did not do this, so I added it for compatability

I think it’s out of reach.
AutoCAD changes the CWD to the .ARX path before On_kInitAppMsg
GetModuleFileName is always the same as std::filesystem::current_path
https://en.cppreference.com/w/cpp/filesystem/current_path

My guess Autodesk does this to attempt to resolve satellite .DLL files.

If you are searching for files, there’s:
Ed.Core.findFile(...)
Ed.Core.findTrustedFile(...)
Db.HostApplicationServices().findFile()	

The last one has an overload that sill start the search from a database location

**Comment by gswifort** [comment]
It seems that this is not an important enough problem to seek a solution.
This can be done directly from Python using the `sitecustomize.py` module, which Python imports during loading (`Py_Initialize`).

```
project/
├── src/
│   └── ...
└── venv/
     └── Lib/
         └── site-packages/
             ├── ...
             └── sitecustomize.py 

```

```py
# sitecustomize.py

import os
import sys
from pathlib import Path

BASE_DIR = Path(__file__).parent


def main():
     project_dir = BASE_DIR.parent.parent.parent
     sys.path.insert(0, str(project_dir))
     os._origin_cwd = str(project_dir)


main()
```

the above code will add the project directory to `sys.path` and create the `os._origin_cwd` variable with the same path.
Of course, the above code makes sense when using a virtual environment and assuming that the virtual environment directory is located directly in the project directory next to the code directory.
